### PR TITLE
[RW-4437][risk=no] Track sidebar state in local storage

### PR DESCRIPTION
**Pull request by dolbeew**
Creates a local storage item to indicate whether the sidebar is open or closed. This way, the sidebar is open automatically only the first time a user logs in and enters a workspace. If the user re-enters a workspace or refreshes the page, it will get the sidebar state from local storage.

Also added a box-shadow in the sidebar so it's clearer that there is content being covered up.

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] I have run and tested this change locally

### CODE REVIEW

File: ui/src/app/pages/workspace/workspace-wrapper/component.ts [hunk 1]
```diff
@@ -58,6 +58,12 @@ export class WorkspaceWrapperComponent implements OnInit, OnDestroy {
   }
 
   ngOnInit(): void {
+    if (!!localStorage.getItem('sidebarState')) {
```
Comment (ericsong) [inline ui/src/app/pages/workspace/workspace-wrapper/component.ts]: nit: pull out the `sidebarState` value into a constant. I think we're (loosely) following the convention of using capital letters for local storage keys. 

Also, prefixing the key value with `WORKSPACE` may be helpful as we're planning to add the side bar component to the notebooks page.

File: ui/src/app/pages/workspace/workspace-wrapper/component.ts [hunk 2]
```diff
@@ -58,6 +58,12 @@ export class WorkspaceWrapperComponent implements OnInit, OnDestroy {
...
+      this.sidebarOpen = localStorage.getItem('sidebarState') === 'open';
+    } else {
+      this.sidebarOpen = true;
```
Comment (ericsong) [inline ui/src/app/pages/workspace/workspace-wrapper/component.ts]: can we use `setSidebarState` here?

File: ui/src/app/pages/workspace/workspace-wrapper/component.ts [hunk 3]
```diff
@@ -67,6 +73,7 @@ export class WorkspaceWrapperComponent implements OnInit, OnDestroy {
           this.setHelpContent();
           if (this.helpContent !== 'reviewParticipantDetail') {
             this.sidebarOpen = false;
+            localStorage.setItem('sidebarState', 'closed');
```
Comment (ericsong) [inline ui/src/app/pages/workspace/workspace-wrapper/component.ts]: also here.

nit - IMO, it's easier to maintain the syncing to localStorage if its done in a one place (the setter) and the rest of the file just uses the setter and doesn't have to worry about localStorage.

File: ui/src/app/pages/workspace/workspace-wrapper/component.ts (hunk 1 above)
Comment (dolbeew) [reply ui/src/app/pages/workspace/workspace-wrapper/component.ts]: Okay, now I see this convention used in the `signed-in` component also. Fixed.

File: ui/src/app/pages/workspace/workspace-wrapper/component.ts (hunk 2 above)
Comment (dolbeew) [reply ui/src/app/pages/workspace/workspace-wrapper/component.ts]: Done.

File: ui/src/app/pages/workspace/workspace-wrapper/component.ts (hunk 3 above)
Comment (dolbeew) [reply ui/src/app/pages/workspace/workspace-wrapper/component.ts]: Agreed, much cleaner this way.

**Pull request merged**